docs/topics/classes.md

```diff
@@ -1,263 +1,587 @@
+interleaved with property initializers:
 
 ```kotlin
-class Customer public @Inject constructor(name: String) { /*...*/ }
+//sampleStart
+// Class with a primary constructor that initializes name and age
+class Person(val name: String, var age: Int) {
+    // First initializer block
+    init {
+        // Runs first when an instance is created
+        println("Person created: $name, age $age.")
+        // Person created: John, age 30.
+    }
+
+    // Second initializer block
+    init {
+        // Runs after the first initializer block
+        if (age < 18) {
+            println("$name is a minor.")
+        } else {
+            println("$name is an adult.")
+            // John is an adult.
+        }
+    }
+}
+
+fun main() {
+    // Creates an instance of the Person class
+    Person("John", 30)
+}
+//sampleEnd
 ```
+{kotlin-runnable="true"}
+
+You can use the primary constructor parameters in the initializer blocks. For example, in the code above, the first and second initializers use
+the `name` and `age` parameters from the primary constructor. 
+
+A common use of `init` blocks is data validation. For example, by calling the [`require` function](https://kotlinlang.org/api/core/kotlin-stdlib/kotlin/require.html):
 
-Learn more about [visibility modifiers](visibility-modifiers.md#constructors).
+```kotlin
+class Person(val age: Int) {
+    init {
+        require(age > 0, "age must be positive")
+    }
+}
+```
 
 ### Secondary constructors
 
-A class can also declare _secondary constructors_, which are prefixed with `constructor`:
+In Kotlin, a secondary constructor is an additional constructor that a class can have beyond its primary constructor. 
+Secondary constructors allow you to provide different ways to initialize an instance when the primary constructor alone 
+isn't enough.
 
-```kotlin
-class Person(val pets: MutableList<Pet> = mutableListOf())
+Secondary constructors are useful for [Java interoperability](java-to-kotlin-interop.md) and when
+you need multiple ways to initialize a class.
+
+To declare a secondary constructor, use the `constructor`
+keyword inside the class body with the constructor parameters within parentheses. Use curly braces with the constructor logic inside:
 
-class Pet {
-    constructor(owner: Person) {
-        owner.pets.add(this) // adds this pet to the list of its owner's pets
+```kotlin
+// Class header with a primary constructor that initializes name and age
+class Person(val name: String, var age: Int) {
+
+    // Secondary constructor that takes a String age
+    // and converts it to an integer
+    constructor(name: String, age: String) : this(name, age.toIntOrNull() ?: 0) {
+        println("$name created with converted age: $age")
+        // Bob created with converted age: 8
     }
 }
+
+fun main() {
+    // Uses the secondary constructor with a String age
+    Person("Bob", "8")
```

Comment:
```suggestion
    Person("Bob", "8")
    // Bob created with converted age: 8
```
I think it makes more sense to add the printed statement here for the constructor.